cmd/webhook_v2.go

```diff
@@ -60,25 +60,27 @@ func newWebhookV2Cmd() *cobra.Command {
 				os.Exit(1)
 			}
 
+			if err := o.resOpts.Validate(); err != nil {
+				log.Fatalf("non-valid resource provided: %v", err)
+			}
+			res := o.resOpts.NewResource(projectInfo.Repo, projectInfo.Domain)
```

Comment:
So, is note the resourceOptionals creating the model? `o.resOpts.NewResource`
Then, I think that would make more sense to keep the resource model as attribute of resource optional instead of. 
Also, why pass the `projectInfo.Repo, projectInfo.Domain`?. IMO: It is info that should be in the resOptionals.